### PR TITLE
feat: Added output to the `avm/res/storage/storage-account` module

### LINKED ISSUE #2994: [AVM Module Issue]: Storage Account module does not return private endpoint network interface IDs
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/storage/storage-account

### (Optional) Module Version

_No response_

### Description

Storage Account module does not return private endpoint network interface IDs. I need to get IDs of network interfaces to enable diagnostic settings to make the deployment complaint with ISO 27001:2013 (policy: "Audit diagnostic setting for selected resource types")

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by msl0**
## Description
The version of the private endpoint module has been bumped to allow expose IDs of private endpoint network interfaces.
Added `privateEndpointsNetworkInterfaceIds` output
Fixes #2994 
Closes #2994 

## Pipeline Reference

| Pipeline |
| -------- |
[avm.res.storage.storage-account](https://github.com/msl0/bicep-registry-modules/actions/workflows/avm.res.storage.storage-account.yml)

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [X] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [X] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [X] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [X] Update to documentation

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] I have run `Set-AVMModule` locally to generate the supporting module files.
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by msl0** [comment]
FYI @sanderaernouts

**Comment by msl0** [comment]
@AlexanderSehr I'm a new contributor. Is there anything else I should do before my PR is reviewed?

**Comment by AlexanderSehr** [comment]
> @AlexanderSehr I'm a new contributor. Is there anything else I should do before my PR is reviewed?

Hey @msl0,
thank you for your contribution. Not per se - the module owners, @fblix & @ktremain should review your PR and provide you with feedback, if any :)
The only thing I'd ask from the top of my head is to please run the pipeline ones including the deploy step so that we can ensure the tests will definitely work in the AVM environment. Only if all tests pass when the PR is merged, the updated module would be published, so we have to make sure everything checks out :)

**Comment by ktremain** [comment]
hi @msl0 I noticed that your pipeline run that you linked above hasn't had the deploy task run (it's listed as skipped) - I assume this is likely missing the 'Azure test environment' setup in your repo (ref: https://azure.github.io/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/#1-setup-your-azure-test-environment) 

I'd be happy to merge in your PR as soon as it has a successful deploy run.

**Comment by msl0** [comment]
Hi @AlexanderSehr @ktremain thank you for the feedback. I will do what you suggest today and let you know when it is finished

**Comment by msl0** [comment]
@fblix, @ktremain
I had some errors related to the non-existing resources and the deployment not being found in Deploy [max], but after pulling changes from main branch and resolving conflicts the last attempt to run the tests was successful
I also had a merge conflict in the version file. In the meantime version 0.13 was created, but I left version 0.12 which I created. Please let me know if it's ok, because I'm making such a change for the first time in this project

**Comment by msl0** [comment]
@fblix @ktremain I don't know if you saw my previous comment, I would like to kindly remind you about feedback

**Comment by msl0** [comment]
The original link to the test results also includes the deployment tests after the last run
https://github.com/msl0/bicep-registry-modules/actions/runs/10455981572

**Comment by matebarabas** [comment]
> The original link to the test results also includes the deployment tests after the last run https://github.com/msl0/bicep-registry-modules/actions/runs/10455981572
> 
> FYI @AlexanderSehr @ktremain @fblix @matebarabas

Hi @msl0, thanks for your patience! FYI, I just reached out to @fblix and @ktremain internally, drawing their attention to this thread.

**Comment by fblix** [comment]
Hi @msl0 big sorry for the delayed response, appearently the github notifications based on mentions don't seem to work for me.
I've added this PR to my watchlist, so I get notified upon a change.

All in all it looks good. One request though: Can you update the e2e tests that currently use private endpoints to also use your changed parameters? If I see it correctly e.g. the max test case still uses the params `privateDnsZoneResourceIds` instead of `privateDnsZoneGroupConfigs`. Could you update the respective test cases and run your pipeline again?

Thanks for your contribution!

File: avm/res/storage/storage-account/main.bicep [hunk 1]
```diff
@@ -700,6 +699,11 @@ output location string = storageAccount.location
 @description('All service endpoints of the deployed storage account, Note Standard_LRS and Standard_ZRS accounts only have a blob service endpoint.')
 output serviceEndpoints object = storageAccount.properties.primaryEndpoints
 
+@description('The array of networ interfaces IDs associated with private endpoints.')
+output privateEndpointsNetworkInterfaceIds array = [
+  for (privateEndpoint, index) in (privateEndpoints ?? []): storageAccount_privateEndpoints[index].outputs.networkInterfaceIds
+]
```
Comment (AlexanderSehr) [inline avm/res/storage/storage-account/main.bicep:705]: Just to align with the output that @krbar implemented in already most other modules where the PE implementation was lifted up to latest. Does not require a re-test, but a re-run of the `Set-AVMModule` script to make sure the readme & template are up to date.

Nothing else to add. Great work @msl0 💪 

```suggestion
@description('The private endpoints of the Storage Account.')
output privateEndpoints array = [
  for (pe, i) in (!empty(privateEndpoints) ? array(privateEndpoints) : []): {
    name: storageAccount_privateEndpoints[i].outputs.name
    resourceId: storageAccount_privateEndpoints[i].outputs.resourceId
    groupId: storageAccount_privateEndpoints[i].outputs.groupId
    customDnsConfig: storageAccount_privateEndpoints[i].outputs.customDnsConfig
    networkInterfaceIds: storageAccount_privateEndpoints[i].outputs.networkInterfaceIds
  }
]
```
```suggestion
@description('The array of networ interfaces IDs associated with private endpoints.')
output privateEndpointsNetworkInterfaceIds array = [
  for (privateEndpoint, index) in (privateEndpoints ?? []): storageAccount_privateEndpoints[index].outputs.networkInterfaceIds
]
```
```suggestion
@description('The array of networ interfaces IDs associated with private endpoints.')
output privateEndpointsNetworkInterfaceIds array = [
  for (privateEndpoint, index) in (privateEndpoints ?? []): storageAccount_privateEndpoints[index].outputs.networkInterfaceIds
]
```

File: avm/res/storage/storage-account/main.json [hunk 2]
```diff
@@ -4963,6 +5063,16 @@
       },
       "value": "[reference('storageAccount').primaryEndpoints]"
     },
+    "privateEndpointsNetworkInterfaceIds": {
```
Comment (AlexanderSehr) [inline avm/res/storage/storage-account/main.json:5066]: Both this file and the readme are currently not up to date with the `main.bicep`

Comment (msl0) [reply avm/res/storage/storage-account/main.json]: I executed Set-AVMModule to fix this

**Comment by msl0** [comment]
Thank you all for your support! I performed the tests one more time to make sure that after the last changes everything is still ok - it's green
https://github.com/msl0/bicep-registry-modules/actions/runs/10715318835

**Review by AlexanderSehr — APPROVED** [review]

**Comment by AlexanderSehr** [comment]
Great work @msl0 and thanks for adding the 'last minute changes'. @fblix given that your last request was for the PE update which looks good, I'll go ahead and merge the PR. 💪

**Pull request merged**